src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Math.round(Double.valueOf(s)) : Long.valueOf(s);
```

Comment:
actually, i tried to make the behavior same as mysql.
Double.valueOf("1.8").longValue() = 1
wherease
Math.round(Double.valueOf("1.8")) = 2 which is exactly same as it would be in mysql.
IMHO, it is better to make 1.8 -> 2 instead of 1.